### ISSUE TITLE
MediaReplayEnginePluginHelper - '@Decorator.cleanup_tmp_dir()' can have unintended consequences

### ISSUE THREAD

**Issue by deadcore** [awslabs/aws-media-replay-engine#17]
Afternoon folks,

It would be great to get your input on something we're experiencing. While we've been developing a selection of plugins & testing them locally we've discovered that the decorator function `@Decorator.cleanup_tmp_dir()` and the function `download_media()` when utilised may have unexpected behaviour on systems

**Function in question**
https://github.com/awslabs/aws-media-replay-engine/blob/d5ae4f57a5c3e01e874a84b06cc668bd7ce64c5c/source/lib/MediaReplayEnginePluginHelper/MediaReplayEnginePluginHelper/__init__.py#L344

**The usage**
https://github.com/awslabs/aws-media-replay-engine/blob/d5ae4f57a5c3e01e874a84b06cc668bd7ce64c5c/source/lib/MediaReplayEnginePluginHelper/MediaReplayEnginePluginHelper/__init__.py#L519

## Expected Behaviour
When cleaning up begins, it executes as expected and remove any temporary files associated with application's logical context

## Actual Behaviour
The decorator attempts to delete the entire `/tmp` directory including those files outside of the applications boundaries. This is especially apparent on Mac

## Proposal
At the moment, we patch the `MediaReplayEnginePluginHelper/__init__.py` by hand to use a `./.tmp` folder so that it will use a `.tmp` file local to where the python interpreter is executed, often the root of the project.

Before going head first into a change, we thought it would be best first to discuss how best the MRE team would see best fit to address this issue. My immediate knee jerk I can see there are 2 strong possible candidates:

#### Proposal 1
Update `def download_media` to pass the `path` into the decorator. I'm not entirely sure if this would actually be possible to extract out from the containing function...

#### Proposal 2
Remove `path` from `def download_media` and set them to be the same. Something like this: 
```python
@Decorator.cleanup_tmp_dir("/tmp/aws-media-replay-engine/media/")
def download_media(self):
    path = "/tmp/aws-media-replay-engine/media/"
```

**Comment by araviraj-aws** [comment]
Hi @deadcore , this is interesting! I assume the cleanup decorator with `/tmp` path becomes an issue only during local testing but not in Lambda's ephemeral storage (since MediaReplayEnginePluginHelper is intended to be built as a layer for Lambda). Nevertheless, I think it makes sense to use a dedicated folder under `/tmp` for MRE specific functions (such as download_media) and modify the cleanup decorator to clean only this folder so that it doesn't affect the rest of the `/tmp` directory. We'll look into addressing this as soon as possible and thanks for your patience!

**Comment by araviraj-aws** [comment]
@deadcore: We just released MRE v2.9.0 which addresses this issue. Please feel free to reopen if you have any additional questions. Thanks again for your patience!
